src/foam/nanos/crunch/lite/CapablePayloadStatusDAO.js

```diff
@@ -0,0 +1,86 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.crunch.lite',
+  name: 'CapablePayloadStatusDAO',
+  extends: 'foam.dao.ProxyDAO',
+
+  javaImports: [
+    'foam.core.ContextAwareAgent',
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.ArraySink',
+    'foam.dao.DAO',
+    'foam.nanos.crunch.Capability',
+    'foam.nanos.crunch.CapabilityJunctionStatus',
+    'foam.nanos.crunch.CrunchService',
+    'foam.nanos.crunch.lite.Capable',
+    'foam.nanos.crunch.lite.CapableAdapterDAO',
+    'foam.nanos.crunch.lite.CapablePayload',
+
+    'foam.nanos.logger.Logger',
+
+    'java.util.List',
+    'java.util.Arrays',
+
+    'static foam.mlang.MLang.*',
+    'static foam.nanos.crunch.CapabilityJunctionStatus.*'
+  ],
+
+  methods: [
+    {
+      name: 'put_',
+      javaCode: `
+        Logger  logger = (Logger) x.get("logger");
+
+        CapableAdapterDAO tempPayloadDAO = (CapableAdapterDAO) x.get("capablePayloadDAO");
+        Capable capableTarget = tempPayloadDAO.getCapable();
+        var payloadDAO = (DAO) capableTarget.getCapablePayloadDAO(x);
+
+        CapablePayload payload = (CapablePayload) obj;
+
+        CapabilityJunctionStatus defaultStatus = PENDING;
+
+        try {
+          payload.validate(x);
+        } catch ( IllegalStateException e ) {
+          logger.warning(e);
+          return obj;
```

Comment:
Assuming exception generating an intercept occurs on validate above. 
If so add comments, if not, simply returning obj seems odd.

Reply:
Done.